File: src/Entity/Recolte.php
<?php

namespace App\Entity;

use App\Repository\RecolteRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: RecolteRepository::class)]
class Recolte
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: "date")]
    #[Assert\NotBlank(message: "La date de récolte est requise.")]
    #[Assert\Type(type: "\DateTimeInterface", message: "La date de récolte doit être valide.")]
    private ?\DateTimeInterface $dateRecolte = null;

    #[ORM\Column(type: "float")]
    #[Assert\NotBlank(message: "La quantité est requise.")]
    #[Assert\Positive(message: "La quantité doit être un nombre positif.")]
    private ?float $quantite = null;

    #[ORM\Column(length: 50)]
    #[Assert\NotBlank(message: "La qualité est requise.")]
    #[Assert\Choice(choices: ["Excellente", "Bonne", "Moyenne", "Médiocre", "Mauvaise"], message: "Choisissez une qualité valide.")]
    private ?string $qualite = null;

    #[ORM\Column(type: "float")]
    #[Assert\NotBlank(message: "Le prix unitaire est requis.")]
    #[Assert\Positive(message: "Le prix unitaire doit être un nombre positif.")]
    private ?float $prixUnitaire = null;

    #[ORM\ManyToOne(targetEntity: Culture::class)]
    #[ORM\JoinColumn(nullable: false)]
    #[Assert\NotBlank(message: "La culture est requise.")]
    private ?Culture $culture = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDateRecolte(): ?\DateTimeInterface
    {
        return $this->dateRecolte;
    }

    public function setDateRecolte(?\DateTimeInterface $dateRecolte): static
    {
        $this->dateRecolte = $dateRecolte;
        return $this;
    }

    public function getQuantite(): ?float
    {
        return $this->quantite;
    }

    public function setQuantite(?float $quantite): static
    {
        $this->quantite = $quantite;
        return $this;
    }

    public function getQualite(): ?string
    {
        return $this->qualite;
    }

    public function setQualite(string $qualite): static
    {
        $this->qualite = $qualite;
        return $this;
    }

    public function getPrixUnitaire(): ?float
    {
        return $this->prixUnitaire;
    }

    public function setPrixUnitaire(?float $prixUnitaire): static
    {
        $this->prixUnitaire = $prixUnitaire;
        return $this;
    }

    public function getCulture(): ?Culture
    {
        return $this->culture;
    }

    public function setCulture(?Culture $culture): static
    {
        $this->culture = $culture;
        return $this;
    }
}

File: src/main/java/Agriwise/entities/Recolte.java
package Agriwise.entities;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Recolte {
    private int id;
    private Date dateRecolte;  // Using java.util.Date instead of String
    private float quantite;
    private String qualite;
    private float prixUnitaire;
    private Culture culture;

    public Recolte() {
    }

    // Getters and Setters

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Date getDateRecolte() {
        return dateRecolte;
    }

    public void setDateRecolte(Date dateRecolte) {
        this.dateRecolte = dateRecolte;
    }

    public float getQuantite() {
        return quantite;
    }

    public void setQuantite(float quantite) {
        this.quantite = quantite;
    }

    public String getQualite() {
        return qualite;
    }

    public void setQualite(String qualite) {
        this.qualite = qualite;
    }

    public float getPrixUnitaire() {
        return prixUnitaire;
    }

    public void setPrixUnitaire(float prixUnitaire) {
        this.prixUnitaire = prixUnitaire;
    }

    public Culture getCulture() {
        return culture;
    }

    public void setCulture(Culture culture) {
        this.culture = culture;
    }

    // Method to validate quality choices
    public static List<String> getQualiteChoices() {
        List<String> choices = new ArrayList<>();
        choices.add("Excellente");
        choices.add("Bonne");
        choices.add("Moyenne");
        choices.add("Médiocre");
        choices.add("Mauvaise");
        return choices;
    }
}
